### ISSUE TITLE
Finish connecting OrganizationGithubAppInstallations

### ISSUE THREAD

**Issue by joshsmith** [code-corps/code-corps-api#845]
# Problem

- [ ] Add relationship of the requesting organization to the GithubAppInstallation
- [ ] Allow for simultaneous creation of the OrganizationGithubAppInstallation and the GithubAppInstallation when creating on that API endpoint

**Comment by begedin** [comment]
This is not completely clear to me.

From that UI you've posted

![image](https://user-images.githubusercontent.com/1925540/28154299-e2c28198-67a9-11e7-8724-10e0c63b8842.png)

 "Install on github" will create a `GithubAppInstallation` belonging to the current user, the current project and the current organization as `requesting_organization`. The way I see it, with the project belonging to the organization, we can already get a `requesting_organization` from there, so this can just be a feature of the `GithubAppInstallation` view if we even need it.

"Connect to Code Corps" will create an `OrganizationGithubAppInstallation` all on it's own. The related `GithubAppInstallation` will have already existed.

With that in mind, it seems like we do not actually need a `requesting_organization` relationship, nor do we need to create an `OrganizationGithubAppInstallation` alongside the `GithubAppInstallation`.

If my assumptions are true, then the actual to-do list here would be

- [x] add `OrganizationGithubAppInstallation#create` endpoint
- [x] add `GithubAppInstallation#create` endpoint
- [ ] add `requesting_organization`  "relationship" to the `GithubAppInstallaiton` view - potentially not needed at all

Basically, this issue might be closed already.

**Comment by joshsmith** [comment]
One thing I was clarifying here is that we want to create an `OrganizationGithubAppInstallation` alongside the `GithubAppInstallation` for the purposes of not having an intermediate flicker in the UI. Assuming the creation goes through, the model might get loaded for the latter before the former is created, meaning it will show up as an "Unconnected installation" briefly before it becomes a "Connected installation".

I expect us for the unconnected and connected installations to show an intermediate loading state where we're "waiting for GitHub" or something.

Alternatively, we could not show _anything_ in the UI by preventing a user from even seeing installations that are in a particular state.

I'm not sure how I feel about this either way.

**Comment by begedin** [comment]
This would be the first time we're creating multiple records with a single post. I'd rather we did it as a promise chain on the client, in that case, since the reason we're doing it both at once is due to how the client UI is set up. 

Unlike here, it would not be the first time we're doing multiple steps in chain on the client. Stripe had a few cases where we did something along those lines.

That being said, I don't feel strongly about it.

**Comment by joshsmith** [comment]
Agree on promise chaining so as long as it doesn't make the UI flicker. It shouldn't, right?

**Comment by begedin** [comment]
@JoshSmith It can be done so there's no flicker.

**Comment by joshsmith** [comment]
This seems like it can be closed, unless I'm misunderstanding.
